### PR TITLE
🏠 Add street address and seasonality to Entry Details page

### LINKED ISSUE #77: Entry location and seasonality
## Description

Add location and seasonality to entry details:

image

Full entry details page for context:

image

Closes #47 

## Todo

- Reverse geocode latitude and longitude of location to produce a street
  - Don't do this logic in the actual component, import a helper function to do this
  - You prob want to move search/locationInfo.js to general utils, and add your helper function there
- Convert season_start and season_end, if no_season is true, to month names
  - Same here: the EntryDetails component is already on the large side
  - Maybe move ACCESS_TYPE, formatISOString, formatSeasonality to entry/textFormatters.js
- Consider creating a separate component (kind of like ResourceList) for the location and seasonality
- Clicking on the location (the address or the icon) should center the map view on the lat and lng of the location itself
image
  - After clicking on the address, a pin should appear over the location:
  image
  - Push /map on desktop too
  - You can do this by adding a field in SearchContext called selectedLocation), passing that to Map
  - So EntryDetails needs to set selectedLocation with the entire location object
  - MapPage has a new useEffect on selectedLocation that's pretty similar to the one on searchViewport, except it does `setView(fitContainerBounds(getGeolocationBounds({ bounds of the location })`
  - Map adds a boolean prop called selected to `<Location>` if `location.id === selectedLocation`
  - Style Location.js to support having a pin over it, if selected is true

### PR DESCRIPTION
**Pull request by code-arman**
## Status:
:rocket: Ready

## Description
Adds location and seasonality to entry details. Also set up styling for the pin to show up when the location is clicked (which will be worked on in another PR)

Fixes #77 

## Todos

- Fix svg width on entries such as the one at ISR

## Screenshots
Screen Shot 2021-04-21 at 2 30 45 PM

### CODE REVIEW

File: src/components/entry/EntryDetails.js [hunk 1]
```diff
@@ -1,36 +1,52 @@
...
-const ACCESS_TYPE = {
-  0: "On lister's property",
-  1: 'Received permission from owner',
-  2: 'Public property',
-  3: 'Private but overhanging',
-  4: 'Private property',
-}
+const IconBesideText = styled.div`
+  display: flex;
+  font-family: inherit;
```
Comment (Qrtn) [inline src/components/entry/EntryDetails.js:28]: Do you need font family inherit?

File: src/components/entry/EntryDetails.js [hunk 2]
```diff
@@ -78,29 +95,41 @@ const Description = styled.section`
...
   }, [id])
 
   const _handleAddressClick = () => {
-    // TODO: handle address click
-    console.log('Map Button Clicked')
+    history.push('/map')
+    const tempViewState = {
+      center: { lat: locationData.lat, lng: locationData.lng },
+      zoom: 18,
+    }
```
Comment (Qrtn) [inline src/components/entry/EntryDetails.js:130]: Can you change the getGeolocationBounds behavior to this as well, then use one function?

File: src/components/ui/ResetButton.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import styled from 'styled-components/macro'
+
+const ResetButton = styled.button`
```
Comment (Qrtn) [inline src/components/ui/ResetButton.js:3]: This is amazing, thank you!

File: src/components/entry/EntryDetails.js [hunk 4]
```diff
@@ -151,6 +180,21 @@ const EntryDetails = () => {
         {typesHeader}
         <Description>
           <p>{locationData.description}</p>
+          <IconBesideText bold onClick={_handleAddressClick} tabIndex={0}>
+            <Map color={theme.secondaryText} size={20} />
+            <LocationText>{address}</LocationText>
+          </IconBesideText>
+          {locationData.season_start && locationData.season_stop && (
```
Comment (Qrtn) [inline src/components/entry/EntryDetails.js:187]: Just checking, did you find any examples of no_season and how that case should be displayed?

File: src/components/entry/EntryDetails.js [hunk 5]
```diff
@@ -1,36 +1,52 @@
...
-  2: 'Public property',
-  3: 'Private but overhanging',
-  4: 'Private property',
-}
+const IconBesideText = styled.div`
+  display: flex;
+  font-family: inherit;
+  color: ${({ theme }) => theme.secondaryText};
+  font-style: normal;
+  font-weight: ${(props) => (props.bold ? 'bold' : 'normal')};
```
Comment (Qrtn) [inline src/components/entry/EntryDetails.js:31]: Add a $ to this prop :P

Comment (Qrtn) [reply src/components/entry/EntryDetails.js:31]: 🤑

File: src/components/entry/EntryDetails.js [hunk 6]
```diff
@@ -151,6 +174,23 @@ const EntryDetails = () => {
         {typesHeader}
         <Description>
           <p>{locationData.description}</p>
+          <IconBesideText bold onClick={handleAddressClick} tabIndex={0}>
+            <Map color={theme.secondaryText} size={20} />
+            <LocationText>{address}</LocationText>
+          </IconBesideText>
+          {locationData.no_season ||
+            (locationData.season_start && locationData.season_stop && (
```
Comment (Qrtn) [inline src/components/entry/EntryDetails.js:182]: nesting's a bit off here. the first ending parentheses should come after `season_stop`

File: src/components/map/Cluster.js [hunk 7]
```diff
@@ -13,13 +15,12 @@ const MIN_CLUSTER_DIAMETER = 30
  */
 const MAX_CLUSTER_DIAMETER = 100
 
-const ClusterContainer = styled.button`
+const ClusterContainer = styled(ResetButton)`
```
Comment (Qrtn) [inline src/components/map/Cluster.js:18]: You can remove cursor:pointer from these styles

File: src/components/ui/IconButton.js [hunk 8]
```diff
@@ -2,15 +2,15 @@ import PropTypes from 'prop-types'
 import { css } from 'styled-components'
 import styled from 'styled-components/macro'
 
-const StyledIconButton = styled.button`
-  cursor: pointer;
+import ResetButton from './ResetButton'
+
+const StyledIconButton = styled(ResetButton)`
```
Comment (Qrtn) [inline src/components/ui/IconButton.js:7]: You can remove cursor:pointer from here too

File: src/utils/viewportBounds.js [hunk 9]
```diff
@@ -1,22 +1,11 @@
 import { getGeocode } from 'use-places-autocomplete'
 
-const CURRENT_LOCATION_VIEWPORT_RADIUS = 0.001
-
-export const getGeolocationBounds = (geolocation) => {
+export const getProperViewState = (latitude, longitude) =>
   // Use fixed viewport around the lat and long of the current location
```
Comment (Qrtn) [inline src/utils/viewportBounds.js:4]: Update this comment -- Use fixed zoom level centered at lat and long

File: src/utils/viewportBounds.js [hunk 10]
```diff
@@ -1,22 +1,11 @@
 import { getGeocode } from 'use-places-autocomplete'
 
-const CURRENT_LOCATION_VIEWPORT_RADIUS = 0.001
-
-export const getGeolocationBounds = (geolocation) => {
+export const getProperViewState = (latitude, longitude) =>
```
Comment (Qrtn) [inline src/utils/viewportBounds.js:3]: `const getZoomedInView = (centerLat, centerLng) =>`

**Review by Qrtn — APPROVED** [review]
Nice work Arman!!

File: src/components/entry/EntryDetails.js (hunk 6 above)
Comment (Qrtn) [reply src/components/entry/EntryDetails.js:182]: Actually just make a variable in the body of entry details. `const hasSeasonality = locationData.no_season || (locationData.season_start && locationData.season_stop)`.

Comment (Qrtn) [reply src/components/entry/EntryDetails.js:182]: `hasSeasonality && JSX`

File: src/components/ui/IconButton.js (hunk 8 above)
Comment (Qrtn) [reply src/components/ui/IconButton.js:7]: Oh sorry if you did already!

File: src/utils/viewportBounds.js (hunk 10 above)
Comment (Qrtn) [reply src/utils/viewportBounds.js:3]: search and replace throughout the project. cmd shift F

File: src/components/entry/EntryDetails.js (hunk 6 above)
Comment (code-arman) [reply src/components/entry/EntryDetails.js:182]: good catch, Quarantine!

File: src/components/ui/IconButton.js (hunk 8 above)
Comment (code-arman) [reply src/components/ui/IconButton.js:7]: uh i think i did cuz i searched for 'pointer' in the file and could not find

File: src/components/entry/EntryDetails.js (hunk 6 above)
Comment (code-arman) [reply src/components/entry/EntryDetails.js:182]: oh damn these additional comments just loaded. glad i saw

File: src/components/map/Location.js [hunk 11]
```diff
@@ -1,26 +1,35 @@
...
 
   &:focus {
     outline: none;
   }
+
+  svg {
+    position: absolute;
+    transform: translate(-50%, -50%);
+    top: -12px;
+    filter: drop-shadow(0px 1px 5px rgba(0, 0, 0, 0.45));
```
Comment (laurenho025) [inline src/components/map/Location.js:30]: Why is this drop shadow color hard coded?

File: src/components/map/Map.js [hunk 12]
```diff
@@ -1,5 +1,6 @@
 import GoogleMap from 'google-map-react'
 import PropTypes from 'prop-types'
+import React from 'react'
```
Comment (laurenho025) [inline src/components/map/Map.js:3]: Remove this react import

File: src/utils/api.ts [hunk 13]
```diff
@@ -59,7 +59,6 @@ export const addLocation = (
       params,
     }),
   )
-
```
Comment (laurenho025) [inline src/utils/api.ts:62]: nit add new line back

File: src/utils/viewportBounds.js [hunk 14]
```diff
@@ -1,22 +1,11 @@
...
-      lng: longitude + CURRENT_LOCATION_VIEWPORT_RADIUS,
-    },
-    sw: {
-      lat: latitude - CURRENT_LOCATION_VIEWPORT_RADIUS,
-      lng: longitude - CURRENT_LOCATION_VIEWPORT_RADIUS,
-    },
-  }
-}
+export const getZoomedInView = (centerLat, centerLng) =>
+  // Update this comment -- Use fixed zoom level centered at lat and long
```
Comment (laurenho025) [inline src/utils/viewportBounds.js:4]: Were you supposed to update this comment?

File: src/components/map/Location.js (hunk 11 above)
Comment (code-arman) [reply src/components/map/Location.js:30]: Jeffraj said it's fine

**Review by laurenho025 — APPROVED** [review]
🚀

**Comment by ezwelty** [comment]
@code-arman

Hmmm, this branch seems to be broken. Opening an entry fails:

e.g. https://falling-fruit-kqa1yl1zp-hack4impact1.vercel.app/entry/3383 (and so https://falling-fruit.vercel.app/entry/3383 after merge)

> react-dom.production.min.js:216 TypeError: Cannot read property 'no_season' of undefined at rt (EntryDetails.js:105)

The error seems to happen before the API (`/locations/{id}.json`) is ever called.

**Pull request merged**